src/chunk.c

```diff
@@ -3278,13 +3190,44 @@ ts_chunk_do_drop_chunks(Hypertable *ht, int64 older_than, int64 newer_than, int3
 	DEBUG_WAITPOINT("drop_chunks_chunks_found");
 
 	if (has_continuous_aggs)
-		ts_chunk_drop_process_invalidations(ht->main_table_relid,
-											older_than,
-											newer_than,
-											chunks,
-											num_chunks);
+	{
+		int i;
 
-	for (; i < num_chunks; i++)
+		/* Exclusively lock all affected chunks and refresh+invalidate the
+		 * dropped regions. Locking prevents further modification of the
+		 * dropped region during this transaction, which allows moving the
+		 * invalidation threshold without having to worry about new
+		 * invalidations while refreshing. */
+		for (i = 0; i < num_chunks; i++)
+		{
+			int64 start = chunks[i].cube->slices[0]->fd.range_start;
+			int64 end = chunks[i].cube->slices[0]->fd.range_end;
+
+			LockRelationOid(chunks[i].table_id, ExclusiveLock);
+
+			Assert(hyperspace_get_open_dimension(ht->space, 0)->fd.id ==
+				   chunks[i].cube->slices[0]->fd.dimension_id);
+
+			/* Refresh all continuous aggregates on the hypertable in the
+			 * region covered by the dropped chunk. Note that we cannot assume
+			 * that all dropped chunks exists in one single contiguous region,
+			 * so we refresh all chunk regions individually rather than the
+			 * region defined by the min and max chunk. An optimization is to
+			 * merge adjecent regions into a larger one for a larger
+			 * refresh. However, such merging needs to account for
+			 * multi-dimensional tables where some chunks have the same
+			 * primary dimension ranges. */
+			ts_cm_functions->continuous_agg_refresh_all(ht, start, end);
```

Comment:
I guess this is kept to be in consistent with previous implementation, but the intention here was not to do it if I am not mistaken?

Reply:
Kept for now, and we might remove later.